### PR TITLE
chore(operator): update ZFS operator with latest csi images

### PR DESCRIPTION
**Pull request by akhilerm**
- update zfs operator with latest snapshotter, provisioner, registrar images

Signed-off-by: Akhil Mohan <akhil.mohan@mayadata.io>

### CODE REVIEW

File: 2.5.0/zfs-operator.yaml [hunk 1]
```diff
@@ -1051,12 +1051,12 @@ spec:
   podInfoOnMount: false
 ---
 
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: (devel)
```
Comment (pawanpraka1) [inline 2.5.0/zfs-operator.yaml:1058]: controller-gen.kubebuilder.io/version: v0.4.0

File: 2.5.0/zfs-operator.yaml [hunk 2]
```diff
@@ -1067,67 +1067,108 @@ spec:
...
   storedVersions: []
+
+
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: (devel)
```
Comment (pawanpraka1) [inline 2.5.0/zfs-operator.yaml:1170]: controller-gen.kubebuilder.io/version: v0.4.0

File: 2.5.0/zfs-operator.yaml [hunk 3]
```diff
@@ -1138,197 +1179,288 @@ spec:
...
   storedVersions: []
----
 
-apiVersion: apiextensions.k8s.io/v1beta1
+---
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: (devel)
```
Comment (pawanpraka1) [inline 2.5.0/zfs-operator.yaml:1462]: controller-gen.kubebuilder.io/version: v0.4.0

Comment (pawanpraka1) [reply 2.5.0/zfs-operator.yaml:1462]: same change in the other places also.

Comment (pawanpraka1) [reply 2.5.0/zfs-operator.yaml:1462]: @akhilerm should we update RC1 and RC2 yamls also?

Comment (akhilerm) [reply 2.5.0/zfs-operator.yaml:1462]: No @pawanpraka1 . RC1/RC2 are not referenced from anywhere and we dont modify the RC yamls.

Comment (akhilerm) [reply 2.5.0/zfs-operator.yaml]: Done.

**Comment by akhilerm** [comment]
> @akhilerm can you get the kubebuilder version from this yaml https://github.com/openebs/zfs-localpv/blob/master/deploy/zfs-operator.yaml. We generate the crds with v0.2.8 kubebuilder and k8s crds(snapshot and other crds) have v0.4.0.

Why is there that difference @pawanpraka1 ?

**Comment by akhilerm** [comment]
@pawanpraka1 Updated it. PTAL.

**Comment by pawanpraka1** [comment]
@akhilerm we have not upgraded the kubebuilder version to v0.4.0 (yet), we are still using v0.2.8 (https://github.com/openebs/zfs-localpv/blob/master/Makefile#L144). So all the crds that we generate will have that label. But the crds that we got from k8s (snapshot, snapshotclass etc) are generated with the latest one.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**